pkgs/jni/lib/src/jobject.dart

```diff
@@ -125,22 +137,36 @@ class JObject {
     reference.release();
   }
 
+  /// Whether this object is of the given [type].
+  ///
+  /// For example:
+  ///
+  /// ```dart
+  /// if (object.isA(JLong.type)) {
+  ///   final i = object.as(JLong.type).longValue;
+  ///   ...
+  /// }
+  /// ```
+  bool isA<T extends JObject?>(JObjType<T> type) {
+    final targetJClass = type.jClass.reference.toPointer();
+    final canBeCasted = Jni.env.IsInstanceOf(reference.pointer, targetJClass);
+    Jni.env.DeleteGlobalRef(targetJClass);
+    return canBeCasted;
+  }
+
   /// Casts this object to another [type].
   ///
   /// If [releaseOriginal] is `true`, the casted object will be released.
+  ///
+  /// Throws [CastError] if this object is not an instance of [type].
   T as<T extends JObject?>(
     JObjType<T> type, {
     bool releaseOriginal = false,
   }) {
-    assert(
```

Comment:
The reason I had used assert here was not to do the type check in release mode. 

Fetching the jclass and calling instanceof might be expensive if we perform a lot of castings. 

Maybe we can have an unsafe flag which is false by default and if true will not do the checking?

Reply:
I think that we should be safe by default.

I'd suggest that we do not offer an unsafe option unless we find a need for it. Put I'd be fine having a `unsafeCast` option, if you'd like me to add one.

Reply:
I'm fine with landing this as is and adding it only if requested.